pkg/feeds/sequence/sequence.go

```diff
@@ -233,12 +233,13 @@ func (f *asyncFinder) at(ctx context.Context, at int64, min int, i *interval, c
 			}
 		}(l)
 	}
+
+	wg.Wait()
 }
 
 func (f *asyncFinder) asyncGet(ctx context.Context, at int64, index uint64) <-chan swarm.Chunk {
-	c := make(chan swarm.Chunk)
+	c := make(chan swarm.Chunk, 1)
```

Comment:
i think this changes the logic substantially and   is incorrect

Reply:
Only change in behavior is that goroutine started at line 242 will finish execution without waiting for other goroutine to read from response channel. There is no need for this goroutine to wait because it's responsability was to provide result  in background. 

Please, explain if I am missing the bigger picture.

Reply:
There is no need for a buffered channel, you can avoid the goroutine leak by using the already passed context:
```golang
select {
	case c <- ch:
	case <-ctx.Done():
}
```

Reply:
Yes that could also do the trick. 

Currently implemented solution, with buffered channel, has these advantages:
- is more robust pattern which could be reused in similar situations even when context is not provided
- goroutine will end early thus freeing it's resources earlier; note that even without intention to fix leaks, pattern without buffered channel will cause goroutines to pile up even when their results it ready, but consumer reading those results is slow.

As a general rule of thumb, whenever there is async method which returns promise like result via channels, it is always best approach to use buffered channel.

Reply:
In general, every goroutine should have an explicit cancellation story, relying on the buffered channel is non-deterministic. In this case, asynchronicity is provided by spinning the goroutine and waiting for the result in select, not by making the channel buffered. Also, if the parent goroutine terminates prematurely due to a quit or context done signal, the child goroutine will not be terminated.

Buffered channels are useful when you want to limit the number of goroutines you will launch, or want to limit the amount of work that is queued up.